app/stac_api/validators.py

```diff
@@ -216,3 +223,306 @@ def validate_asset_multihash(value):
             message=_('Invalid multihash value; %(error)s'),
             params={'error': error}
         )
+
+
+class ValidateSearchRequest:
```

Comment:
Nice! Looks good also from security point of view. (the whole `ValidateSearchRequest` class)